src/pages/SearchPage.js

```diff
@@ -45,69 +45,68 @@ const defaultProps = {
     reports: {},
 };
 
-class SearchPage extends Component {
-    constructor(props) {
-        super(props);
+function SearchPage(props) {
+    const {recentReports: initialRecentReports, personalDetails: initialPersonalDetails, userToInvite: initialUserToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
```

Comment:
I think `getSearchOptions` is a heavy function, would be good to only update the reports, personalDetails and invites only when any of the props changes, to avoid getting the options after every key press

Reply:
I agree with @gedu , `getSearchOptions` is indeed a very heavy function and it's not suitable to have it executing on every single render. You can use `useMemo` to get this initial data only during mounting and pass to your `useState`.